File: src/CadastroIgreja.Domain/Entities/Usuario.cs
namespace CadastroIgreja.Domain.Entities;

public class Usuario
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string NomeCompleto { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Telefone { get; set; }
    public string SenhaHash { get; set; } = string.Empty;
    public Guid IgrejaId { get; set; }
    public Igreja? Igreja { get; set; }
    public bool Ativo { get; set; } = true;
    public DateTime CriadoEm { get; set; } = DateTime.UtcNow;
    public ICollection<UsuarioCargo> UsuarioCargos { get; set; } = new List<UsuarioCargo>();
}

File: frontend/lib/src/models/usuario.dart
class Usuario {
  const Usuario({
    required this.id,
    required this.nomeCompleto,
    required this.email,
    required this.igrejaId,
    required this.ativo,
    this.telefone,
  });

  final String id;
  final String nomeCompleto;
  final String email;
  final String? telefone;
  final String igrejaId;
  final bool ativo;

  factory Usuario.fromJson(Map<String, Object?> json) {
    return Usuario(
      id: json['id'] as String? ?? '',
      nomeCompleto: json['nomeCompleto'] as String? ?? '',
      email: json['email'] as String? ?? '',
      telefone: json['telefone'] as String?,
      igrejaId: json['igrejaId'] as String? ?? '',
      ativo: json['ativo'] as bool? ?? false,
    );
  }
}
